queue.go

```diff
@@ -216,8 +216,8 @@ func (q *Queue) getJobInBucketByID(id uint64, bucketName string) (*Job, error) {
 			log.Printf("Unknown key: %d", id)
 			return nil
 		}
-		job = DecodeJob(jobBytes)
-		return nil
+		job, err = DecodeJob(jobBytes)
+		return err
```

Comment:
You should be able to just return the result from `DecodeJob` here. At least you need to return both `job` and `err`.